### PR TITLE
Allows use of smtps

### PR DESCRIPTION
**Pull request by yjacolin**
More and more smtp service needs authentication to use a SMTP service (SSL,
auth, stattls). So we need to support authenticated smtp, smtps, etc.

### CODE REVIEW

File: c2cgeoportal/lib/email_.py [hunk 1]
```diff
@@ -34,13 +34,23 @@
...
     msg = MIMEMultipart()
     msg["From"] = from_addr
     msg["To"] = ", ".join(to_addrs)
     msg["Date"] = formatdate(localtime=True)
     msg["Subject"] = subject
     msg.attach(MIMEText(body, "plain", "utf-8"))
-    smtp = smtplib.SMTP(smtp_server)
+
+    # Connect to server
+    if 'ssl' in smtp and smtp['ssl']:
```
Comment (arnaud-morvan) [inline c2cgeoportal/lib/email_.py]: if smtp.get('ssl', False) would me better for me

File: c2cgeoportal/lib/email_.py [hunk 2]
```diff
@@ -34,13 +34,23 @@
...
+        smtp = smtplib.SMTP_SSL(smtp['host'])
+    else:
+        smtp = smtplib.SMTP(smtp['host'])
+    if 'user' in smtp and smtp['user']:
```
Comment (arnaud-morvan) [inline c2cgeoportal/lib/email_.py]: same, better to use dict.get with default value

File: doc/integrator/reset_password.rst [hunk 3]
```diff
@@ -29,6 +29,11 @@ And to generate the required e-mail, in the ``vars_<package>.yaml`` file, add th
 
             Sincerely yours
             The GeoMapfish team
-        smtp_server: smtp.example.com
+        smtp:
+          host: smtp.example.com
+          ssl: true
+          user: <username>
+          password: <password>
+          starttls: false
```
Comment (arnaud-morvan) [inline doc/integrator/reset_password.rst]: If the hostname ends with a colon (`:') followed by a number, and
        there is no port specified, that suffix will be stripped off and the
        number interpreted as the port number to use.

File: c2cgeoportal/lib/email_.py (hunk 2 above)
Comment (yjacolin) [reply c2cgeoportal/lib/email_.py]: default value should be '' or null here?

File: doc/integrator/reset_password.rst (hunk 3 above)
Comment (yjacolin) [reply doc/integrator/reset_password.rst]: you mean I need to document this?

**Comment by arnaud-morvan** [comment]
Note that when this will be merged in GMF2, CONST_config-schema.yaml should be updated.

**Comment by yjacolin** [comment]
Note to Arnaud's note ;) : https://github.com/camptocamp/c2cgeoportal/blob/master/c2cgeoportal/scaffolds/update/CONST_config-schema.yaml#L261

File: c2cgeoportal/lib/email_.py (hunk 2 above)
Comment (sbrunner) [reply c2cgeoportal/lib/email_.py]: @arnaud-morvan it's not the same, and it just should be: 
```
if 'user' in smtp:
```

**Comment by sbrunner** [comment]
Is there any sense to be able to put different SMTP servers?
Shouldn't we configure it globally?

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl [hunk 4]
```diff
@@ -1,5 +1,19 @@
...
+==============
+
+1. In the `vars_{{package}}.yaml` file, change:
+
+    smtp_server: smtp.example.com
+
+to
+
+    smtp:
+      host: smtm.example.com
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Use the diff syntax (+/-) a there is done in the rest of the file...

**Comment by arnaud-morvan** [comment]
@sbrunner: is it not configured globally in vars ?

File: c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl (hunk 4 above)
Comment (arnaud-morvan) [reply c2cgeoportal/scaffolds/update/CONST_CHANGELOG.txt_tmpl]: Coould you replace **smtm**.example.com by **smtp**.example.com

File: c2cgeoportal/lib/email_.py [hunk 5]
```diff
@@ -34,13 +34,23 @@
...
     msg = MIMEMultipart()
     msg["From"] = from_addr
     msg["To"] = ", ".join(to_addrs)
     msg["Date"] = formatdate(localtime=True)
     msg["Subject"] = subject
     msg.attach(MIMEText(body, "plain", "utf-8"))
-    smtp = smtplib.SMTP(smtp_server)
+
+    # Connect to server
+    if smtp.get('ssl', false):
```
Comment (fredj) [inline c2cgeoportal/lib/email_.py]: Use `False` instead of `false` (also for the lines above)

Comment (fredj) [reply c2cgeoportal/lib/email_.py]: And the value will be a string, so even if the value is `"False"` in the config file, the test will be true

**Comment by sbrunner** [comment]
Not ad global var, but in the root of configuration, not for each services...

File: c2cgeoportal/lib/email_.py (hunk 5 above)
Comment (sbrunner) [reply c2cgeoportal/lib/email_.py]: In the config file it should be a boolean!

**Comment by arnaud-morvan** [comment]
@sbrunner: can you take a look at the error with travis, this seems related with http://ppa.launchpad.net/stephane-brunner

**Comment by sbrunner** [comment]
You should add 
```yaml
dist: precise
```
in the `.travis.yaml` file https://docs.travis-ci.com/user/reference/overview/ :-)

File: c2cgeoportal/lib/email_.py (hunk 5 above)
Comment (yjacolin) [reply c2cgeoportal/lib/email_.py]: @sbrunner, @fredj  do you mean I have to change something in the vars_xx.yaml file?

**Comment by yjacolin** [comment]
@sbrunner
-  I agree but for GMF 1.6 it is not useful I guess as there is only one feature using it, am I right? But it makes sense for GMF 2.x

File: doc/integrator/reset_password.rst [hunk 6]
```diff
@@ -29,6 +29,15 @@ And to generate the required e-mail, in the ``vars_<package>.yaml`` file, add th
 
             Sincerely yours
             The GeoMapfish team
-        smtp_server: smtp.example.com
+        smtp:
+          host: smtp.example.com:465
+          ssl: true
+          user: <username>
+          password: <password>
+          starttls: false
```
Comment (sbrunner) [inline doc/integrator/reset_password.rst]: incoherent indentation

File: c2cgeoportal/scaffolds/create/vars_+package+.yaml_tmpl [hunk 7]
```diff
@@ -110,7 +110,12 @@ vars:
 
             Sincerely yours
             The GeoMapfish team
-        smtp_server: smtp.example.com
+        smtp:
+          host: smtp.example.com
+          ssl: true
+          user: <username>
+          password: <password>
+          starttls: false
```
Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/vars_+package+.yaml_tmpl]: incoherent indentation

**Comment by sbrunner** [comment]
Then you will change tow times in the migrations ...

**Comment by yjacolin** [comment]
@sbrunner thanks for your feedback. I commited my changes to implement this.

**Pull request merged**